includes/definitions/discovery/scalance.yaml

```diff
@@ -0,0 +1,156 @@
+mib: SN-MSPS-SCX200-MIB
+modules:
+    os:
+        sysDescr_regex: '/SCALANCE (?<hardware>[^,]+), (?<features>[^,]+)/'
+        serial: '.1.3.6.1.4.1.4329.6.3.2.1.1.3.0'
+        version: '.1.3.6.1.4.1.4329.6.3.2.1.1.5.0'
+    processors:
+        data:
+            -
+                # snMspsDiagnosticsUsageTable contains multiple disaparate items with no formal type identifiers, specific OIDs used
```

Comment:
This is why your tests are failing.

Reply:
Thanks, unfortunately I don't have the knowledge yet to resolve this, so will need additional assistance from other to learn what is occurring and how I need to resolve it. Had a couple of attempts on discord to no consensus.
My local unit tests (php 7.4.3) come out OK, so I'm yet to find a local issue I can keep testing against.

Reply:
Thanks for some more tips in discord - that I'm probably seeing a nested array instead of a string value.

This is the table, it works for mempool in the same format - when I try to attack it as a full table I never get a correct CPU value no matter what Index I used:

snmpwalk:
```SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageIndex.0 = INTEGER: 0
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageIndex.1 = INTEGER: 1
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsagePresent.0 = INTEGER: 1
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsagePresent.1 = INTEGER: 1
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageName.0 = STRING: "CPU............................................................................................................................................................................................................................................................"
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageName.1 = STRING: "RAM............................................................................................................................................................................................................................................................"
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageStatus.0 = INTEGER: ok(1)
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageStatus.1 = INTEGER: ok(1)
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageValue.0 = INTEGER: 4
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageValue.1 = INTEGER: 59
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageHighWarning.0 = INTEGER: 0
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageHighWarning.1 = INTEGER: 90
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageHighCritical.0 = INTEGER: 0
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageHighCritical.1 = INTEGER: 98
```
snmptable:
```
SNMP table: SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageTable

 snMspsDiagnosticsUsageIndex snMspsDiagnosticsUsagePresent                                                                                                                                                                                                                                        snMspsDiagnosticsUsageName snMspsDiagnosticsUsageStatus snMspsDiagnosticsUsageValue snMspsDiagnosticsUsageHighWarning snMspsDiagnosticsUsageHighCritical
                           0                             1 "CPU............................................................................................................................................................................................................................................................"                           ok                          21                                 0                                  0
                           1                             1 "RAM............................................................................................................................................................................................................................................................"                           ok                          59                                90                                 98
```

As pointed out - the debug was walking the table and returning 2 values for CPU and Memory, as it's not a table only for CPU or only Memory - let's call it a broken table for these purposes.

```
#### Load disco module processors ####
snMspsDiagnosticsUsageValue.0 = 3
snMspsDiagnosticsUsageValue.1 = 59

#### Load disco module mempools ####
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageValue.0 = 3
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageValue.1 = 59
```
If I then hard-code processor, that works and values/graphs good.

If I hard-code Memory I never get a graph but get correct value in debug.  So assuming we have to use a table for mempool, I tried to exclude the erroneous index 0 CPU value, and I can get the graph displaying.

```
#### Load disco module processors ####
snMspsDiagnosticsUsageValue.0 = 3

#### Load disco module mempools ####
SN-MSPS-SCX200-MIB::snMspsDiagnosticsUsageValue.1 = 59
```

I still don't know how to catch these errors with local unit tests - so submitting again to see if it passes this time.

Reply:
Turns it that was still causing issues with multiple values in mempools, no matter what I tried - left those attempts in the file, but now implemented via php file after following some other examples.